std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoHTTPResponseCache.java

```diff
@@ -0,0 +1,163 @@
+package org.enso.base.enso_cloud;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpHeaders;
+import java.util.List;
+import java.util.Optional;
+import org.enso.base.cache.LRUCache;
+import org.enso.base.cache.ResponseTooLargeException;
+
+/**
+ * EnsoHTTPResponseCache is a cache for EnsoHttpResponse values that respects the cache control HTTP
+ * headers received in the original repsonse to a request.
+ *
+ * <p>It uses LRUCache, so it also puts limits on the size of files that can be requested, and on
+ * the total cache size, deleting entries to make space for new ones. All cache files are set to be
+ * deleted automatically on JVM exit.
+ *
+ * <p>Without caching, EnsoHttpResponse contains an InputStream providing the response data. When
+ * there is a cache hit, this stream reads from the local file storing the cached data. When there
+ * is no cache hit, the InputStream is connected directly to the remote server.
+ */
+public class EnsoHTTPResponseCache {
+  // 1 year.
+  private static final int DEFAULT_TTL_SECONDS = 31536000;
+  private static final long MAX_FILE_SIZE = 2L * 1024 * 1024 * 1024;
+  private static final long MAX_TOTAL_CACHE_SIZE = 20L * 1024 * 1024 * 1024;
```

Comment:
I saw this was 10G and now it's as much as 20G. Why so large limit for this cache?

This cache is kept only during a single session. So I don't know how much processing would someone need to do to gather 10-20GB during a single processing operation.

Btw. should we maybe consider to check `File.getFreeSpace` to see how much space is there left on the device?

1. If our cache can grow up to 20GB, but someone has only 3GB left, it will fail hard with 'no space left on device' at some point. If we check `getFreeSpace`, we could avoid that and be more proactive in cleaning the cache.
2. If someone has little space left, I think it would be good behaviour for the cache to avoid taking over the very last bits of that space - as that may not work well for overall performance of user's system. 
   i. What if we did a check upon initialization to see how much space is left on the device, and set `MAX_TOTAL_CACHE_SIZE` to `min(20GB, [space left] - 1GB)`? (Leaving that 1GB of space left to avoid fully filling up user's disk.)

Reply:
My next cache ticket is to implement environment variables to set these, and I will include a percentage of free space option.

Reply:
Ok